blockmanager.go

```diff
@@ -951,13 +953,17 @@ func (b *blockManager) getCheckpointedCFHeaders(checkpoints []*chainhash.Hash,
 				// match what we know to be the best
 				// checkpoint, then we'll ban the peer so we
 				// can re-allocate the query elsewhere.
-				log.Warnf("Banning peer=%v for invalid "+
-					"checkpoints", sp)
+				peerAddr := sp.Addr()
+				err := b.server.BanPeer(
+					peerAddr,
+					banman.InvalidFilterHeaderCheckpoint,
+				)
+				if err != nil {
+					log.Errorf("Unable to ban peer %v: %v",
+						peerAddr, err)
+				}
 
-				go func() {
-					b.server.BanPeer(sp)
-					sp.Disconnect()
-				}()
+				sp.Disconnect()
```

Comment:
IIRC that goroutines is a work around to prevent a deadlock when attempting to disconnect a peer in this particular context.

Reply:
I think the goroutine was actually only needed for `BanPeer` (see https://github.com/lightninglabs/neutrino/commit/49fcb98e64a6afa0621ff4163aed76a55cf11761), which is no longer an issue as it doesn't go through the server's main goroutine.